docs/en/latest/configuration/config-default.md

```diff
@@ -0,0 +1,1028 @@
+  real_ip_recursive: "off"
+
+  # (string) Defines trusted addresses that are known to send correct replacement addresses. If the special value unix: is specified, all UNIX-domain sockets will be trusted. Trusted addresses may also be specified using a hostname. See here: http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from
+  set_real_ip_from:
+    - 127.0.0.1
+    - "unix:"
+
+  # Add custom shared cache to nginx.conf. Set the cache as "cache-key: cache-size". This configuration is disabled by default.
+  custom_lua_shared_dict:
+
+    # (integer) The size of the ipc shared dictionary.
+    ipc_shared_dict: 100m
+
+  # (boolean) Enables passing of the server name through TLS Server Name Indication extension (SNI, RFC 6066) when establishing a connection with the proxied HTTPS server.
+  proxy_ssl_server_name: true
+
+  # Upstream related configuration
+  upstream:
+
+    # (string) The upstream type. Options are: http, tcp, udp.
+    type: http
+
+    # (integer) The maximum number of idle keep-alive connections to the upstream server that are preserved in the cache of each worker process. If the number of connections exceeds this value, the least recently used connections will be closed.
+    keepalive: 320
+
+    # (integer) The maximum number of request that can be served through one keep-alive connection. If the number of requests exceeds this value, the connection will be closed.
+    keepalive_requests: 1000
+
+    # (integer) Timeout for closing an idle keep-alive connection to the upstream server.
+    keepalive_timeout: 60s
+
+  # (string) Adds the specified charset to the Content-Type response header field. See here: http://nginx.org/en/docs/http/ngx_http_charset_module.html#charset
+  charset: utf-8
+
+  # (integer) The maximum size of the variable hash table.
+  variables_hash_max_size: 2048
+
+  # Lua shared dict configuration
+  lua_shared_dict:
+    internal-status: 10m
+    plugin-limit-req: 10m
+    plugin-limit-count: 10m
+    prometheus-metrics: 10m
+    plugin-limit-conn: 10m
+    upstream-healthcheck: 10m
+    worker-events: 10m
+    lrucache-lock: 10m
+    balancer-ewma: 10m
+    balancer-ewma-locks: 10m
+    balancer-ewma-last-touched-at: 10m
+    plugin-limit-count-redis-cluster-slot-lock: 1m
+    tracing_buffer: 10m
+    plugin-api-breaker: 10m
+    etcd-cluster-health-check: 10m
+    discovery: 1m
+    jwks: 1m
+    introspection: 10m
+    access-tokens: 1m
+    ext-plugin: 1m
+    kubernetes: 1m
+    tars: 1m
+```
+
+### `http_admin_configuration_snippet`
+
+```yaml
+# Add well indented custom Nginx admin server configuration. Please check for conflicts with APISIX snippets.
+http_admin_configuration_snippet: |
```

Comment:
Should we only split the top-level configuration? The indent here is misleading.

Reply:
It looks as if we need the help of this plugin https://github.com/kevin940726/remark-code-import 🤔

Reply:
Currently `http_admin_configuration_snippet` is under `### `http_admin_configuration_snippet`, has a zero indent level, which make it look like a top-level configuration.

Reply:
I just followed what was in the YAML file. please let me know if changes should be done.

Reply:
We can only split the top-level configuration into sections.